src/main/java/seam/SeamCalculator.java

```diff
@@ -31,116 +31,68 @@ public Pixel[][] calculateAndRemoveSeams()
         for (int verticalSeam = 0; verticalSeam < this.verticalSeams.length; verticalSeam++)
         {
             verticalSeams[verticalSeam] = calculateVerticalSeam(newPixels);
-            newPixels = removeVerticalSeam(verticalSeams[verticalSeam], newPixels);
+            Seam seam = verticalSeams[verticalSeam];
+            newPixels = recalculate(removeVerticalSeam(seam, newPixels));
         }
         for (int horizontalSeam = 0; horizontalSeam < this.horizontalSeams.length; horizontalSeam++)
         {
             horizontalSeams[horizontalSeam] = calculateHorizontalSeam(newPixels);
-            newPixels = removeHorizontalSeam(horizontalSeams[horizontalSeam], newPixels);
+            Seam seam = horizontalSeams[horizontalSeam];
+            newPixels = recalculate(removeHorizontalSeam(seam, newPixels));
         }
         return newPixels;
     }
 
-    public Pixel[][] removeVerticalSeam(Seam seam, Pixel[][] pixels)
+    private Pixel[][] recalculate(Pixel[][] pixels)
     {
-        int width = pixels.length - 1;
-        int height = pixels[0].length;
-        Pixel[][] newPixels = new Pixel[width][height];
+        EnergyCalculator energyCalculator = new EnergyCalculator(pixels);
 
-        for (int y = 0; y < height; y++)
-        {
-            for (int x = 0, pixelX = 0; x < width; x++, pixelX++)
-            {
-                if (x == seam.getSeam(y))
-                {
-                    pixelX++;
-                }
-                if (pixelX <= pixels.length - 1)
-                {
-                    newPixels[x][y] = pixels[pixelX][y];
-                }
-            }
-        }
+        // recalculate all energy values
+        energyCalculator.calculateEnergy();
+        energyCalculator.calculateVerticalEnergy();
+        energyCalculator.calculateHorizontalEnergy();
```

Comment:
It's wasteful to recalculate ALL the energies when you are only working on the horizontal or vertical energies.

Reply:
It is wasteful, but the horizontal and vertical energies both change slightly with every seam removal, as I can see now because my current tests after changing this don't pass. Are small differences enough of a reason to sacrifice efficiency?

Reply:
If you are removing vertical seams, it's wasteful to calculate horizontal energies.
Similarly, if you're removing horizontal seams, it's wasteful to calculate the vertical energies, since you're done removing those seams.